src/sqlancer/citus/oracle/tlp/CitusTLPBase.java

```diff
@@ -0,0 +1,210 @@
+package sqlancer.citus.oracle.tlp;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import sqlancer.Randomly;
+import sqlancer.citus.CitusGlobalState;
+import sqlancer.citus.CitusSchema.CitusTable;
+import sqlancer.citus.gen.CitusCommon;
+import sqlancer.postgres.PostgresGlobalState;
+import sqlancer.postgres.PostgresSchema;
+import sqlancer.postgres.PostgresSchema.PostgresColumn;
+import sqlancer.postgres.PostgresSchema.PostgresDataType;
+import sqlancer.postgres.PostgresSchema.PostgresTable;
+import sqlancer.postgres.PostgresSchema.PostgresTables;
+import sqlancer.postgres.ast.PostgresBinaryComparisonOperation;
+import sqlancer.postgres.ast.PostgresBinaryLogicalOperation;
+import sqlancer.postgres.ast.PostgresColumnValue;
+import sqlancer.postgres.ast.PostgresExpression;
+import sqlancer.postgres.ast.PostgresJoin;
+import sqlancer.postgres.ast.PostgresJoin.PostgresJoinType;
+import sqlancer.postgres.ast.PostgresSelect;
+import sqlancer.postgres.ast.PostgresSelect.PostgresFromTable;
+import sqlancer.postgres.gen.PostgresExpressionGenerator;
+import sqlancer.postgres.oracle.tlp.PostgresTLPBase;
+
+public class CitusTLPBase extends PostgresTLPBase {
+
+    HashMap<PostgresTable, Integer> distributedTables;
+    List<PostgresTable> referenceTables;
+    List<PostgresTable> localTables;
+
+    public CitusTLPBase(CitusGlobalState state) {
+        super(state);
+        CitusCommon.addCitusErrors(errors);
+    }
+
+    public PostgresSchema getSchema() {
+        return s;
+    }
+
+    public PostgresTables getTargetTables() {
+        return targetTables;
+    }
+
+    public PostgresExpressionGenerator getGenerator() {
+        return gen;
+    }
+
+    public PostgresSelect getSelect() {
+        return select;
+    }
+
+    public PostgresExpression getPredicate() {
+        return predicate;
+    }
+
+    public PostgresExpression getNegatedPredicate() {
+        return negatedPredicate;
+    }
+
+    public PostgresExpression getIsNullPredicate() {
+        return isNullPredicate;
+    }
+
+    @Override
+    public void check() throws SQLException {
```

Comment:
Since the method actually does not check anything, I'm wondering whether inheriting from `CitusTLPBase` makes sense. Do you think it might be clearer to call this class something like `CitusJoinGenerator`, `CitusTLPSelectGenerator` or so, which has a field `PostgresTLPBase` (if needed)?

Reply:
Since this intercepts a lot of functionalities from PostgresTLPBase, like sets object variables like targetTables etc., how would you recommend going about it? Would you recommend making the signature of certain fields in PostgresTLPBase accessible by other classes so that this class and its methods can modify them?